mlflow/server/js/src/utils/Utils.js

```diff
@@ -107,11 +112,42 @@ class Utils {
         res = <a href={url}>{res}</a>;
       }
       return res;
+    } else if (run.source_type === "NOTEBOOK") {
+      const notebookIdTag = 'mlflow.databricks.notebookID';
+      const webappUrlTag = 'mlflow.databricks.webappURL';
+      const notebookId = tags && tags[notebookIdTag] && tags[notebookIdTag].value;
+      const webappUrl = tags && tags[webappUrlTag] && tags[webappUrlTag].value;
+      if (notebookId && webappUrl) {
+        res = (<a title={run.source_name} href={`${webappUrl}/#notebook/${notebookId}`}>
+          {run.source_name}
+        </a>);
+      }
+      return res;
     } else {
       return res;
     }
   }
 
+  /**
+   * Returns an svg with some styling applied.
+   */
+  static renderSourceTypeIcon(sourceType) {
+    const imageStyle = {
+      height: '20px',
+      position: 'relative',
+      top: '-1px',
+      right: '3px',
+    };
+    if (sourceType === "NOTEBOOK") {
+      return <img title="Source Type: Notebook" style={imageStyle} src={notebookSvg} />;
+    } else if (sourceType === "LOCAL") {
+      return <img title="Source Type: Local" style={imageStyle} src={laptopSvg} />;
+    } else if (sourceType === "PROJECT") {
+      return <img title="Source Type: Project" style={imageStyle} src={projectSvg} />;
```

Comment:
Nit: I'd remove the "Source Type:" in these since they make the tooltip pretty long. Maybe you can just call them "Local Source", "Project" and "Notebook".